### PR TITLE
Enable inference for arbitrary number of instructions

### PR DESCRIPTION
**Pull request by doru1004**
Inferring shapes of operations within a function can be performed in one pass. Switch to a data structure that does not limit the number of operations that we can shape inference on.

### CODE REVIEW

File: src/pass/shape_inference_pass.cpp [hunk 1]
```diff
@@ -36,24 +36,16 @@ class ShapeInferencePass : public mlir::FunctionPass<ShapeInferencePass> {
 
     // Populate the worklist with the operations that need shape inference:
     // these are operations that return a dynamic shape.
-    llvm::SmallPtrSet<mlir::Operation *, 16> op_worklist;
+    llvm::SmallVector<mlir::Operation *, 16> op_worklist;
```
Comment (tjingrant) [inline src/pass/shape_inference_pass.cpp]: What's the reason for changing the container type from Set to a Vector? The change in container semantics here is quite significant.

File: src/pass/shape_inference_pass.cpp [hunk 2]
```diff
@@ -36,24 +36,16 @@ class ShapeInferencePass : public mlir::FunctionPass<ShapeInferencePass> {
...
-      // Find the next operation ready for inference, that is an operation
-      // with all operands already resolved (non-generic).
-      auto nextop = llvm::find_if(op_worklist, returnsDynamicShape);
-      if (nextop == op_worklist.end())
-        break;
-
-      Operation *op = *nextop;
-      op_worklist.erase(op);
-
+    // while (!op_worklist.empty()) {
```
Comment (tjingrant) [inline src/pass/shape_inference_pass.cpp]: Debug code?

File: src/pass/shape_inference_pass.cpp (hunk 1 above)
Comment (doru1004) [reply src/pass/shape_inference_pass.cpp]: SmallPtrSet can't hold more than 32 elements.

**Comment by doru1004** [comment]
op_worklist has been removed since the operations can have their shape inferred as the sequence of operations is traversed.

**Pull request merged**